### PR TITLE
"add init seed"

### PR DESCRIPTION
**Pull request by dzhwinter**
fix https://github.com/PaddlePaddle/Paddle/issues/6220

### CODE REVIEW

File: python/paddle/v2/fluid/initializer.py [hunk 1]
```diff
@@ -302,7 +312,7 @@ class MSRAInitializer(Initializer):
             (https://arxiv.org/abs/1502.01852)
     """
 
-    def __init__(self, uniform=True, fan_in=None, seed=0):
+    def __init__(self, uniform=True, fan_in=None, seed=set_random_seed(0)):
```
Comment (qingqing01) [inline python/paddle/v2/fluid/initializer.py]: Could the user set the seed globally instead of setting the seed at each parameter initializer?

Comment (dzhwinter) [reply python/paddle/v2/fluid/initializer.py]: I didn't get an elegant way to implement it.  Use a global seed seems ugly,

File: python/paddle/v2/fluid/initializer.py [hunk 2]
```diff
@@ -1,7 +1,13 @@
 import framework
 import numpy as np
 
-__all__ = ['Constant', 'Uniform', 'Normal', 'Xavier']
+__all__ = ['Constant', 'Uniform', 'Normal', 'Xavier', 'set_random_seed']
```
Comment (typhoonzero) [inline python/paddle/v2/fluid/initializer.py]: `set_random_seed` seems a private function, if it is, do not expose it.

Comment (dzhwinter) [reply python/paddle/v2/fluid/initializer.py]: fixed.

File: python/paddle/v2/fluid/initializer.py (hunk 1 above)
Comment (dzhwinter) [reply python/paddle/v2/fluid/initializer.py]: fixed

File: python/paddle/v2/fluid/initializer.py [hunk 3]
```diff
@@ -132,6 +132,8 @@ def __call__(self, var, block):
         assert isinstance(var, framework.Variable)
         assert isinstance(block, framework.Block)
         # Initialization Ops should be prepended and not appended
+        if block.program.random_seed != 0 and self._seed == 0:
```
Comment (typhoonzero) [inline python/paddle/v2/fluid/initializer.py]: If we don't consider case that user need to overwrite program default rand seed with 0, just use `if self._seed == 0` is fine, maybe?

Comment (dzhwinter) [reply python/paddle/v2/fluid/initializer.py]: that's true.

**Pull request merged**